stdlib/in_channel.ml

```diff
@@ -63,4 +78,42 @@ let really_input_string ic len =
   | s -> Some s
   | exception End_of_file -> None
 
+let input_all ic =
+  let rec loop buf ofs =
+    let len = Bytes.length buf in
+    if ofs < len then
+      let r = Stdlib.input ic buf ofs (len - ofs) in
+      if r = 0 then
+        Bytes.sub_string buf 0 ofs
+      else
+        loop buf (ofs + r)
+    else begin (* ofs = len *)
+      match Stdlib.input_char ic with
+      | exception End_of_file ->
+          Bytes.unsafe_to_string buf
+      | c ->
+          let new_len = 2 * Bytes.length buf in
+          let new_len =
+            if new_len <= Sys.max_string_length then
+              new_len
+            else if len < Sys.max_string_length then
+              Sys.max_string_length
+            else
+              failwith "In_channel.input_all: cannot grow buffer"
+          in
+          let new_buf = Bytes.create new_len in
+          Bytes.blit buf 0 new_buf 0 ofs;
+          Bytes.set buf ofs c;
+          loop new_buf ofs
+    end
+  in
+  let initial_size =
+    try
+      Stdlib.in_channel_length ic - Stdlib.pos_in ic
+    with Sys_error _ ->
+      0
+  in
+  let initial_size = if initial_size <= 0 then 65536 else initial_size in
+  loop (Bytes.create initial_size) 0
```

Comment:
A bit sad that reading the empty file allocates a 64k buffer.

Reply:
One way to avoid that would be to start with a default initial size of 0 and tweak the resizing code as appropriate (special-case 0 or use 2*len+65536).

Reply:
I amended this by initially allocating an empty buffer and only growing it to 64k if actually needed.